src/managers/TileManager.ts

```diff
@@ -44,46 +46,47 @@ class TileManager {
                 const dto: Nullable<TileDTO> =
                     dtos.find((dto: TileDTO) => dto.id === tile.id) || null;
 
-                let state: TileState = this.identifyTiles(
+                let state: TileState = this.determineState(
                     dto,
-                    session.turnPlayer!,
+                    session.turnPlayer,
                     i,
                 );
                 tile.apply(state, dto);
             }
-            this.getTilesInHand(); //TODO where to save/use
             this.onSync.emit();
         });
     }
 
-    private identifyTiles(
+    private determineState(
         dto: Nullable<TileDTO>,
-        turnPlayer: int,
-        i: int,
+        turnPlayer: Nullable<int>,
+        index: int,
     ): TileState {
+        const InitialAmount: int = this.getInitialAmount(
+            GeneralManager.getPlayers.length,
+        );
+
         if (dto !== null) {
             return TileState.Placed;
         }
-        if (
-            i <
-            turnPlayer +
-                this.getStartOfGameTilesAmount(GeneralManager.getPlayers.length)
-        ) {
+        assert(turnPlayer);
```

Comment:
attention this would also catch int=0 which is evaluated to false. with nullable int do it like this assert x !== null